Revise numerical modeling chapter

astoeriko opened the pull request:
I went over almost the whole chapter to fix typos and small inconsistencies, rephrase somewhat sloppy language and improve formatting. In addition, I made the following changes:

- [x] Revised the introductory text so it better matches the contents of the chapter.
- [x] Revised the Taylor series page to provide better intuition the meaning of Taylor series (also included a new widget)
- [x] Made the exercises on Taylor series reproducible on reloading the page. (My colleague had complained that the exercises change upon reloading the page, so you might not be able to check your solution anymore after working on an exercise.)
- [x] Split the page on IVP into several because it was very long. The section on implicit methods for non-linear ODE now has its own page.
- [x] Revised the section on implicit Euler for non-linear ODE. I chose an example that is closer to reality and revised the text for clarity.
- [x] Added additional small exercises on the pages on single-step and multi-stage methods
- [x] Removed intro page on PDEs, since we do not address PDEs in this chapter anymore

astoeriko committed:
Revise page on Taylor series expansion

astoeriko committed:
Start interactive plot automatically and hide code

astoeriko committed:
Revise widget labels in the interactive plot

astoeriko committed:
Make Taylor series exercises reproducible

Since the exercises use randomly generated numbers for the coefficients,
the exercise would change every time you reload the page. That means
that students could not check their answer for an exercise they worked
on anymore after reloading the page.
Instead, I now provide a fixed set of 4 exercises for each exercise
type. The numbers should now stay the same uplon reloading.

astoeriko committed:
Make exercises responsive to widget

astoeriko committed:
Remove page on PDEs from this chapter

astoeriko committed:
Add missing cell tags and tidy exercise notebook

astoeriko committed:
Merge branch '2025-draft' into revise-numerical-modeling

astoeriko committed:
Start revising non-linear ODE example

astoeriko committed:
Add tag to automatically start live code

astoeriko committed:
Finish revising the section on implicit Euler for non-linear ODE

astoeriko committed:
Fix bioreactor figure

astoeriko committed:
Move section on non-linear ODE to a separate notebook

astoeriko committed:
Revise introductory text

astoeriko committed:
Improve formatting in 1-revision-of-concepts.ipynb

astoeriko committed:
Revise page on boundary value problems

astoeriko committed:
Reinclude notebook outputs

astoeriko committed:
Revise page on multistep and multistage methods

- Made notation in the RK4 equation clearer
- Added an exercise on RK4

astoeriko committed:
Fix typo in exercise

astoeriko committed:
Revise page on single step methods

- fix typos and sloppy language
- remove exercise from section on the explicit Euler methods because it
  just repeated the text
- instead, add an equivalent exercise to the implicit Euler section

astoeriko committed:
Add proper references to the following chapters

astoeriko committed:
Revise intro text on the finite difference method

Mostly reordered the sentences to start with the motivation
instead of with a definition.

astoeriko committed:
Some more small language fixes

astoeriko committed:
Fix formatting in Taylor series exercise

astoeriko committed:
Add note about dark mode on exercise page

astoeriko committed:
Merge branch '2025-draft' into revise-numerical-modeling

astoeriko commented:
My revisions of the numerical modeling chapter are finally ready for review. I have worked on a lot of smaller issues and some medium changes (see my list in the pull request description). The biggest changes are probably on the page on implicit methods for non-linear ODE.

@fmeer @mzijlema : I removed the introductory page on PDEs (previously the last section of this chapter) since we do not treat PDEs anymore here. Do you need this content later in Q2? If so, please review it and move it to your chapter. I renamed the file to `X-starting_with_PDEs.ipynb`.

You can view the built book for this pull request here: https://mude.citg.tudelft.nl/book/revise-numerical-modeling/numerical_methods/overview.html

astoeriko commented:
Don't know why the book fails to build after I merged the main branch, the error messages are not very informative. If I try building locally, I get this error:

```
Could not import extension teachbooks_favourites (exception: No module named 'teachbooks_favourites')
```

@Tom-van-Woudenberg , do you have an idea?

Tom-van-Woudenberg commented:
> Don't know why the book fails to build after I merged the main branch, the error messages are not very informative.

It failed because of this: `Error: HttpError: Deployment request failed for 93c8800ec4b3379ef7346a2cc65216d37ab6a5ab due to in progress deployment. Please cancel 6023b4deba2c5139fda5ebf67145bb1bb456e5e2 first or wait for it to complete.` So another workflow was already updating the server. I'll rerun the workflow so that the red cross is gone ;)

> If I try building locally, I get this error:
>
> ```
> Could not import extension teachbooks_favourites (exception: No module named 'teachbooks_favourites')
> ```

Yes, i've recently updated the `requirements.txt`. If you update your local environment with this new requirements file you should be able to continue

astoeriko commented:
> Yes, i've recently updated the `requirements.txt`. If you update your local environment with this new requirements file you should be able to continue

Thanks, that worked. I think I was confused because the automatic build also did not work, but this was for a completely different reason then. :-)

astoeriko committed:
Two minor edits I forgot to save

astoeriko committed:
Merge branch '2025-draft' into revise-numerical-modeling

This involves small revisions of the page on multistep methods

astoeriko committed:
Try fixing equation formatting

astoeriko committed:
Some more revisions on the Taylor seires page

- Fix some typos and improve formatting
- Better integrate text and exercises in the secion on deriving
  approximations for the first derivative. Previously, the exercises
  just repeated what was already described in the text, which was
  confusing. I removed the exercise on the forward difference and
  central difference, only keeping the text. For the backward
  difference, I converted the text into an exercise.

astoeriko committed:
Fix one more typo

rbjbrinkgreve commented:
Hi Anna,

When I try to edit the below text directly in the book, the code somehow starts below where I want to propose a small change (?)

Anyhow, this is what I wanted to change, but maybe you can do it, please:

***@***.***

Best regards / met vriendelijke groet,
Ronald


From: Anna Störiko ***@***.***>
Sent: Friday, 29 August 2025 16:46
To: TUDelft-MUDE/book ***@***.***>
Cc: Ronald Brinkgreve ***@***.***>; Review requested ***@***.***>
Subject: Re: [TUDelft-MUDE/book] Revise numerical modeling chapter (PR #31)


@astoeriko<https://github.com/astoeriko> requested your review on: #31<https://github.com/TUDelft-MUDE/book/pull/31> Revise numerical modeling chapter.

—
Reply to this email directly, view it on GitHub<https://github.com/TUDelft-MUDE/book/pull/31#event-19413263290>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BLKLZ5HHQH62XRIU2HTGB3T3QBRRFAVCNFSM6AAAAAB4F7UCUWVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJZGQYTGMRWGMZDSMA>.
You are receiving this because your review was requested.Message ID: ***@***.******@***.***>>

astoeriko committed:
More formatting and typos in the Taylor series section

astoeriko committed:
Fix typo that Ronald found

astoeriko commented:
> Hi Anna, When I try to edit the below text directly in the book, the code somehow starts below where I want to propose a small change (?) Anyhow, this is what I wanted to change, but maybe you can do it, please: 

@rbjbrinkgreve I am not sure if I understand your issue with editing the text. But I fixed the typo you highlighted in the picture.

astoeriko committed:
Fix equation formatting in Taylor series exercise

astoeriko committed:
Link back to initial example when discussing Newton's method

astoeriko commented:
@rbjbrinkgreve Is this pull request for the numerical modeling chapter ready to merge from your point of view?

@TUDelft-MUDE/mude-mt I think we also need a review from one of you. Who could volunteer?

rbjbrinkgreve commented:
Dear all,

I reviewed the last bits of the different updated sections on numerical modelling and am quite happy. I think we can merge into the book.

Best regards / met vriendelijke groet,
Ronald

________________________________
Dr. Ronald B.J. Brinkgreve - Associate Professor

[1611821633934]  CEG - Geo-Engineering Section
________________________________

From: Anna Störiko ***@***.***>
Sent: Wednesday, 3 September 2025 15:32
To: TUDelft-MUDE/book ***@***.***>
Cc: Ronald Brinkgreve ***@***.***>; Mention ***@***.***>
Subject: Re: [TUDelft-MUDE/book] Revise numerical modeling chapter (PR #31)

[https://avatars.githubusercontent.com/u/42246615?s=20&v=4]astoeriko left a comment (TUDelft-MUDE/book#31)<https://github.com/TUDelft-MUDE/book/pull/31#issuecomment-3249276118>

@rbjbrinkgreve<https://github.com/rbjbrinkgreve> Is this pull request for the numerical modeling chapter ready to merge from your point of view?

@TUDelft-MUDE/mude-mt<https://github.com/orgs/TUDelft-MUDE/teams/mude-mt> I think we also need a review from one of you. Who could volunteer?

—
Reply to this email directly, view it on GitHub<https://github.com/TUDelft-MUDE/book/pull/31#issuecomment-3249276118>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BLKLZ5F7WJLPMBOTTYPIGFL3Q3UU5AVCNFSM6AAAAAB4F7UCUWVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTENBZGI3TMMJRHA>.
You are receiving this because you were mentioned.Message ID: ***@***.******@***.***>>

Tom-van-Woudenberg commented:
> @TUDelft-MUDE/mude-mt I think we also need a review from one of you. Who could volunteer?

@jialeidng , could you review this?

fmeer committed:
Fix equation that does not render properly

fmeer commented:
This revision deserves an update in de credits. At least a "special thanks", but looking at the changes I think @astoeriko should also be elevated to chapter author, maybe @rbjbrinkgreve as well if this was a joint effort where Anna just took care of the github side of things

fmeer commented:
Many compliments from my side on this revision. Great improvement to the chapter!

fmeer committed:
Merge branch '2025-draft' into revise-numerical-modeling
